src/llvm/abi/avr.cr

```diff
@@ -0,0 +1,82 @@
+require "../abi"
+
+class LLVM::ABI::AVR < LLVM::ABI
+  AVRTINY = StaticArray[
+    "attiny4",
+    "attiny5",
+    "attiny9",
+    "attiny10",
+    "attiny102",
+    "attiny104",
+    "attiny20",
+    "attiny40",
+  ]
+
+  def initialize(target_machine : TargetMachine, mcpu : String? = nil)
+    super target_machine
+
+    # "Reduced Tiny" core devices only have 16 General Purpose Registers
+    if mcpu && AVRTINY.includes?(mcpu)
```

Comment:
Shorter version:

```suggestion
    if mcpu.in?(AVRTINY)
```

Reply:
Note that this is not identical because the truthiness check is missing.
An equivalent shorter version could look like this:
```suggestion
    if mcpu.try &.in?(AVRTINY)
```
I'm happy with either.

Reply:
I'm not sure it's easier to read, but the nilable check isn't needed, so `mcpu.in?(AVRTINY)` would be enough. I prefer `#includes` but I guess `#in?` may read better.